src/irmin/tree.ml

```diff
@@ -650,9 +649,10 @@ module Make (P : Private.S) = struct
             (* Starting from this point the function is expensive, but there is
                no alternative. *)
             cnt.node_val_list <- cnt.node_val_list + 1;
-            let entries = P.Node.Val.list v in
-            (* XXX: A sequence could be used here *)
-            List.for_all (fun (step, _) -> StepMap.mem step um) entries)
+            let entries = P.Node.Val.seq v in
+            Seq.fold_left
+              (fun ok (step, _) -> ok && StepMap.mem step um)
+              true entries)
```

Comment:
Or we could add [`Seq.for_all`](https://github.com/ocaml/ocaml/blob/98c4c0743afe2dad65871c9d527a3e664b90c11f/stdlib/seq.ml#L146-L151) to `import.ml` and use that here for its short-circuiting behaviour.